### PR TITLE
Simple JS+WASM version demo with emscripten

### PR DESCRIPTION
**Pull request by adriweb**
Tested successfully in FF and Chrome.

### CODE REVIEW

File: emscripten/luna-utils.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    var reader = new FileReader();
+    reader.onloadend = function(event) {
+        fileLoaded(event, filename);
+    };
+    reader.readAsArrayBuffer(file);
+};
+
+} // preRun function
+};// Module
```
Comment (Vogtinator) [inline emscripten/luna-utils.js]: Missing newline

File: emscripten/luna.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<!doctype html>
+<html lang="en-us">
+
+<head>
+  <meta charset="utf-8">
+  <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
+  <title>luna-web</title>
+  <script type="text/javascript">
```
Comment (Vogtinator) [inline emscripten/luna.html]: Unobfuscated code into a separate file please

File: emscripten/luna-utils.js (hunk 1 above)
Comment (adriweb) [inline emscripten/luna-utils.js]: done

File: emscripten/luna.html (hunk 2 above)
Comment (adriweb) [inline emscripten/luna.html]: done

File: emscripten/Makefile [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+all: $(OUTPUT).js
+
+%.bc: %.c
+	$(CC) $(CFLAGS) $< -o $@
+
+$(OUTPUT).js: luna.bc zip.bc ioapi.bc DES.bc
+	$(CC) $(CFLAGS) $(LFLAGS) -o $@ $^
+
+clean:
+	$(RM) -r $(OUTPUT).js* $(OUTPUT).data *.bc
```
Comment (Vogtinator) [inline emscripten/Makefile]: Using *.bc here but listing all files seperately above is inconsistent and might cause bugs

Comment (adriweb) [inline emscripten/Makefile]: Well, same thing in the main Makefile, then... I guess I'll fix both.

File: emscripten/luna-utils.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+var Module = { 'preRun': function() {
+
+callLuna = function(luaFileName, tnsFileName, cb)
+{
+    var ret = Module.callMain([luaFileName, tnsFileName]);
```
Comment (Vogtinator) [inline emscripten/luna-utils.js]: This code looks like it got infected by the node.js callback virus. IMO that does not make any sense here, what about the simple linear approach directly in fileLoaded?

File: emscripten/luna-utils.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+var Module = { 'preRun': function() {
+
+callLuna = function(luaFileName, tnsFileName)
+{
+   Module.callMain([luaFileName, tnsFileName]);
```
Comment (Vogtinator) [inline emscripten/luna-utils.js]: Missing return code handling?

Comment (adriweb) [inline emscripten/luna-utils.js]: It turns out `callMain` doesn't return anything.

Comment (Vogtinator) [inline emscripten/luna-utils.js]: Then maybe just run Module["main"] directly? The return code is definitely useful outside of luna

Comment (adriweb) [inline emscripten/luna-utils.js]: done

**Comment by adriweb** [comment]
I've changed some stuff to also be compatible with WebAssembly. The Makefile has been updated too, to make such a target.
Both work fine on Chrome with wasm support enabled, FF Nightly, and Safari (using asm.js fallback if built in Wasm, otherwise, the expected asm.js directly).

File: emscripten/luna-utils.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        }
+    }
+};
+
+fileLoad = function(event)
+{
+    var file = event.target.files[0];
+    if (!file || file.name.substr(-4) !== '.lua')
+    {
+        alert('No .lua script file given');
```
Comment (Vogtinator) [inline emscripten/luna-utils.js]: Misleading error message for the 2nd case

File: emscripten/luna.html [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  <meta charset="utf-8">
+  <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
+  <title>luna-web</title>
+  <script src="FileSaver.js" type="text/javascript"></script>
+</head>
+
+<body>
+
+<div>
+    Create a .tns file from a .lua script: <input type="file" id="luaFile" accept=".lua" onChange="fileLoad(event)"/><br>
```
Comment (Vogtinator) [inline emscripten/luna.html]: This works just fine for Problem1.xml without any modifications, so maybe accept both?

File: emscripten/luna.html [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+</div>
+
+<script type='text/javascript'>
+    var Module = { postRun: [] };
+    window.onerror = function() { };
```
Comment (Vogtinator) [inline emscripten/luna.html]: Why is this here?

File: emscripten/luna-utils.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
+var Module;
+if (!Module) Module = (typeof Module !== 'undefined' ? Module : null) || {};
```
Comment (Vogtinator) [inline emscripten/luna-utils.js]: It's not obvious why this is needed, please add a comment here

File: emscripten/Makefile [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+%.bc: %.c
+	$(CC) $(CFLAGS) $< -o $@
+
+$(OUTPUT).js: $(OBJECTS)
+	$(CC) $(CFLAGS) $(LFLAGS) -o $@ $^
+
+clean:
+	$(RM) -r $(OUTPUT).js* $(OUTPUT).asm.js $(OUTPUT).was* $(OUTPUT).data $(OBJECTS)
+
+.PHONY: all clean
```
Comment (Vogtinator) [inline emscripten/Makefile]: Missing asmjs wasm here

File: emscripten/luna-utils.js (hunk 9 above)
Comment (adriweb) [inline emscripten/luna-utils.js]: That's some code from emscripten directly, but I'll add a comment.

File: emscripten/luna.html (hunk 8 above)
Comment (adriweb) [inline emscripten/luna.html]: old stuff, fixed.

File: emscripten/Makefile (hunk 10 above)
Comment (adriweb) [inline emscripten/Makefile]: fixed.

File: emscripten/luna.html (hunk 7 above)
Comment (adriweb) [inline emscripten/luna.html]: mkay

File: emscripten/luna-utils.js (hunk 6 above)
Comment (adriweb) [inline emscripten/luna-utils.js]: fixed

File: .gitignore [hunk 11]
```diff
@@ -1,3 +1,9 @@
+*.bc
 *.o
+*.wasm
```
Comment (Vogtinator) [inline .gitignore]: Please move this into a new .gitignore inside the emscripten directory
